app/src/main/java/org/oppia/app/options/OptionsFragmentPresenter.kt

```diff
@@ -122,23 +134,43 @@ class OptionsFragmentPresenter @Inject constructor(
     binding: OptionStoryTextSizeBinding,
     model: OptionsReadingTextSizeViewModel
   ) {
+    binding.commonViewModel = viewModel
     binding.viewModel = model
   }
 
   private fun bindAppLanguage(
     binding: OptionAppLanguageBinding,
     model: OptionsAppLanguageViewModel
   ) {
+    binding.commonViewModel = viewModel
     binding.viewModel = model
   }
 
   private fun bindAudioLanguage(
     binding: OptionAudioLanguageBinding,
     model: OptionsAudioLanguageViewModel
   ) {
+    binding.commonViewModel = viewModel
     binding.viewModel = model
   }
 
+  fun setSelectedFragment(selectedFragment: String) {
+    viewModel.selectedFragmentIndex.set(
+      getSelectedFragmentIndex(
+        selectedFragment
+      )
+    )
+  }
+
+  private fun getSelectedFragmentIndex(selectedFragment: String): Int {
+    return when (selectedFragment) {
+      READING_TEXT_SIZE_FRAGMENT -> 0
+      APP_LANGUAGE_FRAGMENT -> 1
+      DEFAULT_AUDIO_FRAGMENT -> 2
+      else -> throw InvalidParameterException()
```

Comment:
Please write some error message too inside this exception like:
`InvalidParameterException ("Not a valid fragment in getSelectedFragmentIndex.")`

Reply:
Done